### DISCUSSION TITLE
Failed to flash: Failed to update deck bcAI:gap8

### DISCUSSION THREAD

**Discussion by machacekch** [bitcraze/discussions#1400] [Q&A] [answered]
I've recently purchased a cf2 and followed the [general getting started guide](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/), installing the clients on win10 with `pip3 install cfclient`. Crazyradio is set up, so is lighthouse positioning, and I've done a few successful flights in the GUI cfclient. All decks have been updated (individually) to the newest firmware.

I'm running into a problem with the AI deck. Following [its getting started guide](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example), flashing the wifi example via radio fails with a non-specific error, likely [here](https://github.com/bitcraze/crazyflie-lib-python/blob/master/cflib/bootloader/__init__.py#L642):

```
>cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w radio://0/80/2M/E7E7E7E7E7
Reset to bootloader mode ...
nRF51 has: sd-s130 and requires None and upgrade provides None. Current bootloader version is [2024.2.0] but upgrade provides [None]
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcLighthouse4, not in the target list
Skipping bcAI:esp, not in the target list
Deck bcAI:gap8, reset to bootloader
Failed to flash: Failed to update deck bcAI:gap8
```
I've tried both the 2022.05 and the 2023.10-rc2 version of the [wifi image streamer .bin](https://github.com/bitcraze/aideck-gap8-examples/releases), both of which show the same error.

Any idea what might be causing the problem? Unlike users in some other threads, I never see any progress-percentage. It just fails immediately. The cf does reset though and the decks are enumerated correctly, so it doesn't seem to be a connection issue.

(Please note I'm on win10; the guide mentions that Ubuntu is required, but I suppose that's just because it describes setting up the development environment further down. Also, I've had the lighthouse deck attached in addition to the AI deck while running the above.)

**Comment by gemenerik** [comment, marked as answer]
That's unfortunate! Can you try without the lighthouse deck and let me know if that works?

**Reply by machacekch** [reply]
Yes indeed, without the LH deck flashing proceeds to 99%, resets the cf2 and gets stuck there. So there might be a secondary issue, but this one's resolved. Thanks!

**Reply by gemenerik** [reply]
Glad to hear that the workaround is helping with the issue. It shouldn't be happening in the first place, and I'll keep an eye on it to see if we can find a more lasting fix.

**Reply by gemenerik** [reply]
Upon further review, I realized that I recently committed a potential fix for this problem in https://github.com/bitcraze/crazyflie-firmware/commit/c2b949702e2a64017f769ad52f39c952359b5cf7. However, I haven't tested it in your specific configuration yet. My apologies for any confusion caused.

Removing the deck every time you want to flash is not a sustainable solution. Can you retry while running the latest master branch of the crazyflie-firmware? I recommend testing this on your end to see if it resolves the issue. If you find that the fix does not work, please let me know, and I will investigate further.

**Reply by machacekch** [reply]
> Removing the deck every time you want to flash is not a sustainable solution. Can you retry while running the latest master branch of the crazyflie-firmware? I recommend testing this on your end to see if it resolves the issue. If you find that the fix does not work, please let me know, and I will investigate further.

I've tried this and instead of `Failed to flash: Failed to update deck bcAI:gap8` it now gets stuck 3%. I've written up a detailed description of what I've tried and the issues observed in #1436. I suggest we continue the discussion there.
